builtin/mainmenu/settings/components.lua

```diff
@@ -126,6 +126,40 @@ end, is_valid_number)
 make.string = make_field(tostring, nil)
 
 
+function make.slider(setting)
+	local steps = tonumber(setting.steps) or 100
+
+	return {
+		info_text = setting.comment,
+		setting = setting,
+
+		get_formspec = function(self, avail_w)
+			local value = core.settings:get(setting.name) or setting.default
+			local scrollbar_value = value/(setting.max - setting.min)*steps - setting.min
+
+			self.resettable = true
+
+			local setting_label = ("label[0,0.25;%s]"):format(
+				setting.readable_name)
+			local scrollbar_options = ("scrollbaroptions[min=0;max=%f;smallstep=1;largestep=10]"):format(
+				steps)
+			local scrollbar_formspec = ("scrollbar[0,0.5;%f,0.4;horizontal;%s;%f]"):format(
+				avail_w - 1, setting.name, scrollbar_value)
+			local formspec = setting_label..scrollbar_options..scrollbar_formspec
+
+			return formspec, 1.1
+		end,
+
+		on_submit = function(self, fields)
+			local raw_value = core.explode_scrollbar_event(fields[setting.name]).value
+			local value = (raw_value/steps) * (setting.max - setting.min) + setting.min
+			core.settings:set(setting.name, tostring(value))
+			return false
```

Comment:
The value should only be changed if the scrollbar is modified

```suggestion
			local event = core.explode_scrollbar_event(fields[setting.name])
			if event.type ~= "CHG" then
				return false
			end
			local raw_value = event.value
			local value = (raw_value/steps) * (setting.max - setting.min) + setting.min
			core.settings:set(setting.name, tostring(value))
			return true
```

Reply:
Reason why mouse dragging stops working is that after returning `true` GUI gets regenerated and this probably causes scrollbar to lose focus.

Reply:
normally regenerating the GUI doesn't make scrollbars loose focus, but for some minetest reason it does in the main menu